### PR TITLE
Added Bookmark feature

### PR DESCRIPTION
**Pull request by ManishShah120**
# Description[about the feature]:
- Added the Bookmarking/Saving feature.
- Added an extra dropdown menu under the profile section.

Fixes: #171 

# Additional Info:
  - Converted the `post-detail` view from a class-based view to a function-based view, so as a result more features and functions can be added to the post-detailed view/page in the future.
  - Added an extra separate page for displaying all the saved/bookmarked blogs for any specific user.
  - Also added feature to remove the previously saved/bookmarked blog-posts.
  - Added a condition for allowing only logged in user to be able to save or bookmarks.
  
# Screenshots/Gif:
BookMark_Feature

### CODE REVIEW

**Comment by ManishShah120** [comment]
@harsh-9in & @Kajol-Kumari , Please have a look.

File: mysite/blog/templates/blog/base.html [hunk 1]
```diff
@@ -41,7 +41,21 @@
...
-              <a class="nav-item nav-link" href="{% url 'profile' %}">Profile</a>
+
+              <ul class="navbar-nav">
+                  <li class="nav-item dropdown">
+                      <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button"
+                          data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Profile
+                      </a>
+                      <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+                          <a class="dropdown-item" href="{% url 'profile' %}">Dashboard</a>
+                          <a class="dropdown-item" href="{% url 'users:favourite_list' %}">Saved Blogs</a>
```
Comment (Kajol-Kumari) [inline mysite/blog/templates/blog/base.html:52]: Please replace `saved Blogs` with `My Favourite blogs`

File: mysite/users/templates/user/favourites.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{% extends 'blog/base.html'%}
+{% load static %}
+
+{% block content %}
+    <h2>Favourites</h2>
```
Comment (Kajol-Kumari) [inline mysite/users/templates/user/favourites.html:5]: Let's keep the heading here as `My Favourite Blogs`

File: mysite/users/urls.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    favourite_add,
+    favourite_list,
+)
+
+app_name = 'users'
+
+urlpatterns = [
+    path('fav/<int:id>/', favourite_add, name='favourite_add'),
+    path('favourites/', favourite_list, name='favourite_list'),
+]
```
Comment (Kajol-Kumari) [inline mysite/users/urls.py:12]: please leave one blank line at the end of the file.

File: mysite/users/templates/user/favourites.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            </div>
+            <p class="article-content">{{ post.content|truncatechars:200|safe }}</p>
+            <hr />
+            <button class="btn btn-light btn-sm disabled" type="submit">❤</button>
+            <button class="btn btn-light btn-sm disabled" type="submit">👎</button>
+            <button class="btn btn-light btn-sm disabled" type="submit">💬</button>
+        </div>
+    </article>
+    {% endfor %}
+{% endblock %}
```
Comment (Kajol-Kumari) [inline mysite/users/templates/user/favourites.html:22]: please left one extra line here.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@ManishShah120 Please incorporate the suggested changes, Rest LGTM :)

**Comment by ManishShah120** [comment]
@Kajol-Kumari , I have resolved the requested changes. Can it be merged now?

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1: 
@harsh-9in this can be merged now!

**Review by harsh-9in — COMMENTED** [review]
@ManishShah120 plz remove conflicts.

**Comment by ManishShah120** [comment]
@harsh-9in, removed the conflict.

**Review by harsh-9in — APPROVED** [review]
@ManishShah120 plz remove conflicts.

**Pull request merged**